openedx/openedx-platform
# feat: Track additional discussion context for better deletion handling [BD-38] [TNL-9673] [BB-5626]

## Description

Tracks a discussion topics's section and subsection in a new context field so that we have access to that information when the topic's section/subsection/unit has been deleted. This is then used when a topic is deleted to append the section and subsection name to the topic title.

ADR: https://github.com/openedx/edx-platform/pull/29928

## Supporting information

- https://openedx.atlassian.net/browse/TNL-9673

## Testing instructions

1. Switch to this PR and make sure the corresponding version of openedx-events is installed: https://github.com/openedx/openedx-events/pull/51
2. Check out he corresponding PR for the discussions MFE: https://github.com/openedx/frontend-app-discussions/pull/109
3. Create a course using the new openedx provider, or switch an existing course to use it. 
4. Check the discussions MFE and see that the topics from the units show up. 
5. Delete a unit, section or subsection. 
6. Refresh the MFE. The topics should now be in an archived section with format "<section>|<subsection>|<unit>"

## Deadline

"None"
